src/main/java/org/folio/rest/impl/PoLinesAPI.java

```diff
@@ -40,28 +55,140 @@ public void getOrdersStoragePoLines(String query, int offset, int limit, String
   @Override
   @Validate
   public void postOrdersStoragePoLines(String lang, org.folio.rest.jaxrs.model.PoLine entity, Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+                                       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     PgUtil.post(POLINE_TABLE, entity, okapiHeaders, vertxContext, PostOrdersStoragePoLinesResponse.class, asyncResultHandler);
   }
 
   @Override
   @Validate
   public void getOrdersStoragePoLinesById(String id, String lang, Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.getById(POLINE_TABLE, PoLine.class, id, okapiHeaders,vertxContext, GetOrdersStoragePoLinesByIdResponse.class, asyncResultHandler);
+                                          Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    PgUtil.getById(POLINE_TABLE, PoLine.class, id, okapiHeaders, vertxContext, GetOrdersStoragePoLinesByIdResponse.class, asyncResultHandler);
   }
 
   @Override
   @Validate
   public void deleteOrdersStoragePoLinesById(String id, String lang, Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.deleteById(POLINE_TABLE, id, okapiHeaders, vertxContext, DeleteOrdersStoragePoLinesByIdResponse.class, asyncResultHandler);
+                                             Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    try {
+      vertxContext.runOnContext(v -> {
+        TxWithId tx = new TxWithId(id);
+        log.info("Delete POLine");
+        startTxWithId(tx)
+          .thenCompose(this::deletePiecesByPOLineId)
+          .thenCompose(this::deletePOLineById)
+          .thenCompose(this::endTxWithId)
+          .thenAccept(result -> {
+            log.info("Preparing response to client");
+            asyncResultHandler.handle(Future.succeededFuture(DeleteOrdersStoragePoLinesByIdResponse.respond204()));
+          })
+          .exceptionally(t -> {
+            endTxWithId(tx)
```

Comment:
Looking at [PostgresClient.rollbackTx(...)](https://github.com/folio-org/raml-module-builder/blob/v24.0.0/domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java#L628) I see that after rollback the connection should be closed.

Reply:
fixed rollback transaction